tools/autograd/gen_variable_type.py

```diff
@@ -988,40 +1085,18 @@ def emit_call(env):
                 call = DISPATCH_TO_NON_VAR_TYPE_WITH_TMP_RETURN_VALUES.substitute(
                     base_type_call=base_type_call)
 
-                call += wrap_output(tie_return_values(), 'tmp')
+                call += wrap_output(tie_return_values, 'tmp')
```

Comment:
I don't know if this is too fussy, but I think the code would be easier to follow if `tie_return_values` was now passed into `emit_call()` explicitly. (Actually FWIW I think having `format_return_variables()` gather those three bits of codegen together might be making the logistics a bit higher-friction than if they were still independent, though ofc factoring to make it/them callable from both tracing and variable is totally the right thing to do, and I see why you brought them together.)

Totally up to you if you want to change any of this - basically just nits.

Reply:
Passed in tie_return_values explicitly.

Yeah - I grouped them together to share these util methods...
```
    arguments = declaration['arguments']
    returns = declaration['returns']
    func = declaration['derivative']
    name = declaration['name']
    inplace = declaration['inplace']
    is_out_fn = name.endswith('_out')
    modifies_arguments = inplace or is_out_fn
    returns_void = len(returns) == 0
```